### ISSUE TITLE
Can't determine RecordFormat for sealed traits with only case objects (enums) when located in package object

### ISSUE THREAD

**Issue by CasperKoning** [sksamuel/avro4s#152]
Consider:
```scala
package example

sealed trait Bar
case object Baz extends Bar
case object Bazz extends Bar

case class Foo(bar: Bar)

object Example extends App {
  import com.sksamuel.avro4s.RecordFormat

  val genericRecord = RecordFormat[Foo].to(Foo(Baz))

  println(genericRecord)
}
```
This runs fine.

Now, when putting the domain classes in a package object instead in a separate file (in example/package.scala):
```scala
package object domain {
  sealed trait Bar
  case object Baz extends Bar
  case object Bazz extends Bar

  case class Foo(bar: Bar)
}
```
and keep the rest of the application the same (in example/Example.scala):
```scala
object Example extends App {
  import com.sksamuel.avro4s.RecordFormat

  val genericRecord = RecordFormat[Foo].to(Foo(Baz))

  println(genericRecord)
}
```
the application fails to compile with the errors:
```
Error:(6, 35) could not find implicit value for parameter fromRecord: com.sksamuel.avro4s.FromRecord[example.Foo]
  val genericRecord = RecordFormat[Foo].to(Foo(Baz))
```
and
```
Error:(6, 35) not enough arguments for method apply: (implicit toRecord: com.sksamuel.avro4s.ToRecord[example.Foo], implicit fromRecord: com.sksamuel.avro4s.FromRecord[example.Foo])com.sksamuel.avro4s.RecordFormat[example.Foo] in object RecordFormat.
Unspecified value parameter fromRecord.
  val genericRecord = RecordFormat[Foo].to(Foo(Baz))
```

Interestingly, this problem only seems to occur when one of the fields of my case classes in my package object is a sealed trait with only case objects as implementors, whilst it also works fine in case of other non-enum fields.

**Comment by CasperKoning** [comment]
The same also happens with the following two files (basically just with an ordinary object):
- `example/Domain.scala`:

   ```scala
    package example
    
    object Domain {
      sealed trait Bar
      case object Baz extends Bar
      case object Bazz extends Bar
      case class Foo(bar: Bar)
    }   
    ```
- `example/Example.scala`

    ```scala
    package example
    
    object Example extends App {
      import Domain._
    
      import com.sksamuel.avro4s.RecordFormat
    
      val genericRecord = RecordFormat[Foo].to(Foo(Baz))
    
      println(genericRecord)
    }
    ```

Compiler error:
```
could not find implicit value for parameter fromRecord: com.sksamuel.avro4s.FromRecord[example.Domain.Foo]
```

**Comment by mahe-ymc** [comment]
We also are affected by this issue. @sksamuel did you make progress here already?

**Comment by sksamuel** [comment]
Fixed in master. Will be in 1.8.4 which will be released today.

**Comment by 2chilled** [comment]
Fix got reverted because of lack of 2.11 support. A Version which works for 2.12. can be found [here](https://github.com/2chilled/avro4s)

**Comment by sksamuel** [comment]
I cannot get this to work. There's subtleties at play here which are not the result of ADT issues in the old 1.9 branch. When using Shapeless's `Reify` it doesn't seem to find the case objects in the package object.

**Comment by sksamuel** [comment]
@2chilled I've dropped support for 2.11 from 2.0 onwards, if you want to have another go at getting this working, you're a better man than me.

**Comment by sksamuel** [comment]
Can you try the 3.0.x snapshot from
https://oss.sonatype.org/content/repositories/snapshots/com/sksamuel/avro4s
This should be fixed and is available for 2.11 and 2.22
